### DISCUSSION TITLE
How to load a plaintext/json asset into a string

### DISCUSSION THREAD

**Discussion by andoco** [bevyengine/bevy#3140] [Q&A] [answered]
I'd like to load a json asset file into string for parsing by the serde library. It looks like text files aren't currently supported for loading via the `AssetServer`, so would I need to implement an `Asset` type and `AssetLoader` for text/json files? Or perhaps there's a better alternative approach?

**Comment by hymm** [comment, marked as answer]
The way to do it is a custom asset loader. There are some published package that you can use for inspiration here https://bevyengine.org/assets/#helpers. There's also an example in the bevy repo.

If you're just trying to hack something together and don't care about blocking rendering for a bit, you can also just do blocking file operations from inside of a system.

**Comment by pianosaurus** [comment]
There are plugins available for this now. The place to look is https://bevyengine.org/assets/#asset-loading, and specifically https://github.com/NiklasEi/bevy_common_assets handles json for you (as well as several other formats).
